readthedocs/search/documents.py

```diff
@@ -48,6 +48,15 @@ class ProjectDocument(RTDDocTypeMixin, Document):
 
     modified_model_field = 'modified_date'
 
+    def get_queryset(self):
+        """
+        Don't include delisted projects.
+        This will also break in-doc search for these projects,
```

Comment:
Hrm... this seems kind of bad. @stsewd thoughts here on if there's a nicer way to implement this? It seems like we probably want to filter them out dynamically in the search results, as well as in the indexing 🤔   

We should definitely be excluding *spam* projects in a similar way, if we aren't already.

Reply:
There is an open issue about that at: https://github.com/readthedocs/readthedocs.org/issues/9899.

Doing the filter here will exclude the project from indexing new builds/changes, I think we should also delete the current index, we could do this in the post save() method, checking if the project was marked as spam or delisted. How to detect the change, that depends on from where we mark the project as delisted, from a form should be easy. And we should also check for the other way too, if there was a mistake, we should detect the change and re-build the index.

Reply:
Thanks for the elaboration @stsewd :+1: 

Does it make sense to still exclude delisted projects (and spam projects) in this query?

Reply:
> Does it make sense to still exclude delisted projects (and spam projects) in this query?

Yeah, we don't want to index new changes/builds. Of course, we can also make this check before triggering the tasks, but I think it's fine to have the filter in the query.

Reply:
> . How to detect the change, that depends on from where we mark the project as delisted, from a form should be easy.

Btw this happens in the Django admin.

Reply:
I think I got confused here.

I'm 👍🏼 on not showing results for delisted and spam projects in search results. However, if possible, we shouldn't not break their in-doc search for delisted projects. I don't care about spam 😄

Reply:
For spam projects, I think we are find removing their search index, for delisted projects, we could have an additional field in our search index that we can filter on.

Reply:
I've included removal of projects and pages of projects marked with `is_spam`. It's a bit more complicated with spam score, since it either requires to copy out code from the readthedocsext extension or to add a new helper function. I'll defer to a later action here. I'll post an issue in readthedocsext, as I think the right solution might be to maintain a spam flag on the project, rather than calculating spam scores dynamically...

Reply:
Ha! @humitos has already been here!

I have added some thoughts in the existing issue: https://github.com/readthedocs/readthedocs-ext/issues/461

I don't think we should expand the scope of this PR, fixing the greater issue with spam project indexing is even more important and requires the same type of fix and already has an issue.

Reply:
Agreed on moving this to another PR, and getting this in 👍